### PR TITLE
Publish with IPFS using a pinning service guide

### PR DESCRIPTION
**Pull request by 2color**
# Describe your changes

Add a new quickstart guide on publishing to IPFS. For the first example use web3.storage, but be clear that there are also other great pinning services.

Fixes: https://github.com/ipfs/ipfs-docs/issues/1421

Preview: https://bafybeif6w6ckulsybmjltcv34xaj54lxssxj6slfll3szgokw3bsokz6em.on.fleek.co/quickstart/publish/

### CODE REVIEW

**Comment by mishmosh** [comment]
Very exciting! A few high-level comments:

- I see this is filed under "Quickstarts" but it's not represented in Navigation — is the idea to replace the "Guides" section with "Quickstarts", and have this be one of the entries under that?
- The web UI is good for the absolute beginner, but given our primary audience of developers, maybe this should use a curl example instead (or in parallel)? It's a bigger leap from web UI to using this in an application, whereas the API example might be more clear.
- The "Gateways" section — I'd be inclined to present that section from the action of "Fetching your published file" and _then_ discuss the options of gateways vs. local node (e.g. IPFS Desktop app and/or embedded node like Audius). The example could use gateways.

**Comment by ElPaisano** [comment]
Hey @2color, I seem to have overwritten your latest with https://github.com/ipfs/ipfs-docs/pull/1490/commits/a12a4c0111f1217a181e5cdf38f0aa18858b7c8f. I was trying to just suggest changes from my review - sorry about that, the intention was not to force push my requested changes on top of yours. I can revert this

**Review by ElPaisano — COMMENTED** [review]
Hey @2color, I seem to have overwritten your latest with https://github.com/ipfs/ipfs-docs/pull/1490/commits/a12a4c0111f1217a181e5cdf38f0aa18858b7c8f. I was trying to just suggest changes from my review - sorry about that, the intention was not to force push my requested changes on top of yours. I can revert this

(that being said, some of the changes were things like adding in the sidebar menu, so lmk if you want to keep parts of the commit, preview https://bafybeieaxkrgw4dkgekyxo7vjntujsn3ajtmptkhq3qiml3now4nqkfb2m.on.fleek.co/quickstart/publish/)

**Comment by 2color** [comment]
> I see this is filed under "Quickstarts" but it's not represented in Navigation — is the idea to replace the "Guides" section with "Quickstarts", and have this be one of the entries under that?

The idea is to have _quickstart_ guides in the **Get Started** section of the docs. These would be different to the longer guides and their goal is to introduce concepts more "softly".

> The web UI is good for the absolute beginner, but given our primary audience of developers, maybe this should use a curl example instead (or in parallel)? It's a bigger leap from web UI to using this in an application, whereas the API example might offer a more direct hop to understanding how to use this within an app.

100%. Once we land this one, the next step would be to take the same structure of this guide and create more versions:
- With terminal / curl 
- With the S3 SDK/CLI and Filebase
- With one of the SDKs 

> The "Gateways" section — I'd be inclined to present that section from the action of "Fetching your published file" and then discuss the options of gateways vs. local node (e.g. IPFS Desktop app and/or embedded node like Audius). The example could use gateways.

I agree. I'll adapt it to follow this structure.

> Will this replace any of the content under Guides > Manage files?

That whole section needs to be revisited. This first quickstart guide wouldn't replace any of the existing content.

However, as a next step, I'd revisit http://localhost:8080/how-to/pin-files and move some of the conceptual pinning content to concepts

File: docs/quickstart/publish.md [hunk 1]
```diff
@@ -5,88 +5,105 @@ description: 'learn about pinning services and the lifecycle of content-addresse
...
+- [https://ipfs.io/ipfs/bafybeicn7i3soqdgr7dwnrwytgq4zxy7a5jpkizrvhm5mv6bgjd32wm3q4](https://ipfs.io/ipfs/bafybeicn7i3soqdgr7dwnrwytgq4zxy7a5jpkizrvhm5mv6bgjd32wm3q4)
+- [https://cloudflare-ipfs.com/ipfs/bafybeicn7i3soqdgr7dwnrwytgq4zxy7a5jpkizrvhm5mv6bgjd32wm3q4](https://cloudflare-ipfs.com/ipfs/bafybeicn7i3soqdgr7dwnrwytgq4zxy7a5jpkizrvhm5mv6bgjd32wm3q4)
+- [https://gateway.pinata.cloud/ipfs/bafybeicn7i3soqdgr7dwnrwytgq4zxy7a5jpkizrvhm5mv6bgjd32wm3q4](https://gateway.pinata.cloud/ipfs/bafybeicn7i3soqdgr7dwnrwytgq4zxy7a5jpkizrvhm5mv6bgjd32wm3q4)
 
-Congratulations, you have successfully **pinned** a file to IPFS!
+## Prerequisites
 
-### What's a CID
+- A free [web3.storage](https://web3.storage/) account.
+- The [following image](./images/welcome-to-IPFS.jpg), downloaded and saved on your computer:
```
Comment (2color) [inline docs/quickstart/publish.md:74]: Btw, using a relative link here for a non-rendered image doesn't work. Hence the prior use of an absolute URL.

**Comment by 2color** [comment]
>Hey @2color, I seem to have overwritten your latest with https://github.com/ipfs/ipfs-docs/commit/a12a4c0111f1217a181e5cdf38f0aa18858b7c8f. I was trying to just suggest changes from my review - sorry about that, the intention was not to force push my requested changes on top of yours. I can revert this

No worries. Thanks for the suggestions. I've incorporated some of them.

**Comment by mishmosh** [comment]
> The idea is to have _quickstart_ guides in the **Get Started** section of the docs. These would be different to the longer guides and their goal is to introduce concepts more "softly".

Gotcha. Nice!

**Comment by ElPaisano** [comment]
@2color following up on this. Looking at https://bafybeif6w6ckulsybmjltcv34xaj54lxssxj6slfll3szgokw3bsokz6em.on.fleek.co/quickstart/publish/#the-lifecycle-of-data-in-ipfs. Do you need any missing content filled in or a review or anything from me to get this shipped? LMK

**Review by ElPaisano — COMMENTED** [review]
@2color reviewed. These are very nicely written guides! I'm a fan of the structure 

To hopefully save you some work, I did my review and made edits accordingly in [1f69375](https://github.com/ipfs/ipfs-docs/pull/1490/commits/1f69375ca5b5e288ca1342593b9d5f335f9faed4). In that commit, I also wrote out the summary for retrieve.md, and added one next step (you may want to add more?).

Also, based on your `<!-- omit from toc -->` comments, I removed the sidebar menu thingy using `sidebarDepth: 0` on `retrieve.md` and `publish.md` - not sure if that's what the note was talking about tho, lmk

LMK if you want me to revert this and just comment instead.

Anyways, I'd say this is ready to go, so if you're good with it, I'm good with merging it

**Comment by 2color** [comment]
Thanks for the review and edits. I'll go over all of this again.

> Also, based on your  comments, I removed the sidebar menu thingy using sidebarDepth: 0 on retrieve.md and publish.md - not sure if that's what the note was talking about tho, lmk

Actually, this was just for the autogenerated TOC that my VSCode plugin generates for Markdown. I

**Comment by 2color** [comment]
Btw @ElPaisano, the reason I use `> Note` rather than the `:::callout` elements is because I find that the callouts pull the attention away too much due to their colour. I'm fine leaving them here to stay consistent with other pages, but we should probably revisit this and lighten the colour at some point.

**Comment by ElPaisano** [comment]
> Btw @ElPaisano, the reason I use `> Note` rather than the `:::callout` elements is because I find that the callouts pull the attention away too much due to their colour. I'm fine leaving them here to stay consistent with other pages, but we should probably revisit this and lighten the colour at some point.

Sounds good, Ill make an issue in the docs board around this

**Review by ElPaisano — APPROVED** [review]
Let's send it

**Pull request merged**